### PR TITLE
chore: upgrade workflows and Dockerfile to use ubuntu 24 LTS

### PR DESCRIPTION
**Pull request by justinhynes**
**Run JavaScript tests locally with Karma**

There is work being done on a fix to get Karma to run in CI. Until then, however, contributors are required to run these tests locally.

- [ ] Make sure you are inside the devstack container
- [ ] Run `make test-karma`
- [ ] All tests pass

### CODE REVIEW

File: .github/workflows/migrations-mysql8-check.yml [hunk 1]
```diff
@@ -10,11 +10,10 @@ on:
 jobs:
   check_migrations:
     name: check migrations
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
```
Comment (justinhynes) [inline .github/workflows/migrations-mysql8-check.yml:13]: I don't think it matters which version of Ubuntu we use in this workflow so I've switched it to use `ubuntu-latest`, which matches the OS version [used in the CI workflow](https://github.com/openedx/credentials/blob/master/.github/workflows/ci.yml). `Ubuntu-latest` is used when running code coverage, unit tests, and quality/linting checks for Credentials today.

**Comment by justinhynes** [comment]
I built the Docker image locally and tested it with Devstack: 
- Verified APIs that ingest/create grades from edx-platform (via `notify_credentials`)
- Verified APIs that ingest/create credentials from edx-platform (via `notify_credentials`)
- Verified APIs that ingest course and program metadata from Discovery (via `copy_catalog`)
- Verified rendering of program certificate using example certificate (http://localhost:18150/credentials/example)
- Verified APIS that feed data to the Learner Record MFE (rendered list of programs, rendered list of courses in specific program, shared program record, rendered shared program record, downloaded program record grades CSV)

**Comment by justinhynes** [comment]
Image build logs:
```
(credentials) ubuntu@ip-172-31-43-154:~/edx-repos/credentials$ docker build -t test_update .
[+] Building 48.7s (34/34) FINISHED                                                                                                                                         docker:default
 => [internal] load build definition from Dockerfile                                                                                                                                  0.0s
 => => transferring dockerfile: 4.86kB                                                                                                                                                0.0s
 => WARN: FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 97)                                                                                                       0.0s
 => [internal] load metadata for docker.io/library/ubuntu:noble                                                                                                                       0.1s
 => [internal] load .dockerignore                                                                                                                                                     0.0s
 => => transferring context: 164B                                                                                                                                                     0.0s
 => [base  1/26] FROM docker.io/library/ubuntu:noble@sha256:8a37d68f4f73ebf3d4efafbcf66379bf3728902a8038616808f04e34a9ab63ee                                                          0.0s
 => [internal] load build context                                                                                                                                                     1.2s
 => => transferring context: 128.61MB                                                                                                                                                 1.2s
 => CACHED [base  2/26] RUN apt-get update &&     apt-get install -y software-properties-common &&     apt-add-repository -y ppa:deadsnakes/ppa && apt-get update &&     apt-get upg  0.0s
 => CACHED [base  3/26] RUN python3.11 -m venv /edx/app/credentials/venvs/credentials                                                                                                 0.0s
 => CACHED [base  4/26] RUN pip install nodeenv                                                                                                                                       0.0s
 => CACHED [base  5/26] RUN nodeenv /edx/app/credentials/nodeenvs/credentials --node=20.17.0 --prebuilt                                                                               0.0s
 => CACHED [base  6/26] RUN npm install -g npm@10.x.x                                                                                                                                 0.0s
 => CACHED [base  7/26] RUN locale-gen en_US.UTF-8                                                                                                                                    0.0s
 => CACHED [base  8/26] RUN useradd -m --shell /bin/false app                                                                                                                         0.0s
 => CACHED [base  9/26] RUN wget https://github.com/facebook/watchman/releases/download/v2023.11.20.00/watchman-v2023.11.20.00-linux.zip                                              0.0s
 => CACHED [base 10/26] RUN unzip watchman-v2023.11.20.00-linux.zip                                                                                                                   0.0s
 => CACHED [base 11/26] RUN mkdir -p /usr/local/{bin,lib} /usr/local/var/run/watchman                                                                                                 0.0s
 => CACHED [base 12/26] RUN cp watchman-v2023.11.20.00-linux/bin/* /usr/local/bin                                                                                                     0.0s
 => CACHED [base 13/26] RUN cp watchman-v2023.11.20.00-linux/lib/* /usr/local/lib                                                                                                     0.0s
 => CACHED [base 14/26] RUN chmod 755 /usr/local/bin/watchman                                                                                                                         0.0s
 => CACHED [base 15/26] RUN chmod 2777 /usr/local/var/run/watchman                                                                                                                    0.0s
 => CACHED [base 16/26] WORKDIR /edx/app/credentials/credentials                                                                                                                      0.0s
 => CACHED [base 17/26] COPY requirements/production.txt /edx/app/credentials/credentials/requirements/production.txt                                                                 0.0s
 => CACHED [base 18/26] COPY requirements/pip_tools.txt /edx/app/credentials/credentials/requirements/pip_tools.txt                                                                   0.0s
 => CACHED [base 19/26] RUN pip install -r requirements/pip_tools.txt                                                                                                                 0.0s
 => CACHED [base 20/26] RUN pip install -r requirements/production.txt                                                                                                                0.0s
 => CACHED [base 21/26] RUN mkdir -p /edx/var/log                                                                                                                                     0.0s
 => [base 22/26] COPY . /edx/app/credentials/credentials                                                                                                                              1.3s
 => [base 23/26] RUN make pull_translations                                                                                                                                           1.4s
 => [base 24/26] RUN npm install --no-save                                                                                                                                           11.4s
 => [base 25/26] RUN webpack --config webpack.config.js                                                                                                                              12.5s
 => [base 26/26] RUN chown -R app:app /edx/app/credentials/credentials/credentials/static                                                                                             0.7s
 => [dev 1/3] RUN pip install -r /edx/app/credentials/credentials/requirements/dev.txt                                                                                               15.1s
 => [dev 2/3] RUN make pull_translations                                                                                                                                              1.6s
 => [dev 3/3] RUN touch ../credentials_env                                                                                                                                            0.2s
 => exporting to image                                                                                                                                                                2.9s
 => => exporting layers                                                                                                                                                               2.9s
 => => writing image sha256:8081d27cd9274536b2052e4434cdcf676ede4fb441774d1a0f46345e45a731a7                                                                                          0.0s
 => => naming to docker.io/library/test_update
```

File: Dockerfile [hunk 2]
```diff
@@ -30,9 +30,9 @@ ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 # Create Node env
 RUN pip install nodeenv
 ENV NODE_ENV=/edx/app/credentials/nodeenvs/credentials
-RUN nodeenv $NODE_ENV --node=18.17.1 --prebuilt
+RUN nodeenv $NODE_ENV --node=20.17.0 --prebuilt
```
Comment (deborahgu) [inline Dockerfile:33]: nice bonus; we hadn't ticketed this!

File: Dockerfile [hunk 3]
```diff
@@ -30,9 +30,9 @@ ENV PATH="$VIRTUAL_ENV/bin:$PATH"
...
 ENV PATH="$NODE_ENV/bin:$PATH"
-RUN npm install -g npm@9.x.x
+RUN npm install -g npm@10.x.x
```
Comment (deborahgu) [inline Dockerfile:35]: I assume this is accurate; when I went looking to see if it was necessary, i couldn't confirm it but I am sure you are correct.

**Review by deborahgu — APPROVED** [review]

File: Dockerfile (hunk 3 above)
Comment (justinhynes) [reply Dockerfile:35]: When I used NVM to install Node 20 in my devstack instance, it automagically installed NPM 10.8.2 alongside it. I figured Node was best paired with a version of NPM similar to what NVM provides. 

Taking a quick glance, it looks like Node 20.17.x+ requires NPM 10.8.2+. 

I _could_ make it a bit more specific to `npm@10.8.x`?

File: Dockerfile (hunk 2 above)
Comment (justinhynes) [reply Dockerfile:33]: haha, figured since we were in here and already making these changes in other places... 😂

**Pull request merged**